src/effects/effects_base/internal/effectexecutionscenario.cpp

```diff
@@ -82,22 +83,23 @@ muse::Ret EffectExecutionScenario::doPerformEffect(au3::Au3Project& project, con
     secs_t t1;
     bool isSelection = false;
 
+    const auto numSelectedClips = selectionController()->selectedClips().size();
+
     {
         //! NOTE Step 1.2 - get effect
         effect = effectsProvider()->effect(effectId);
 
-        //! NOTE Step 1.3 - check selection
-        trackedit::ClipKeyList selectedClips = selectionController()->selectedClips();
-        if (selectedClips.size() == 1) {
+        if (numSelectedClips > 1 && !effectsProvider()->supportsMultipleClipSelection(effectId)) {
+            return make_ret(Err::EffectMultipleClipSelectionNotSupported);
+        }
+
+        if (selectionController()->hasSelectedClips()) {
+            // If multiple clips are selected, we have checked that the effect supports it, in which case these global time boundaries shouldn't be relevant.
+            // If this is just a single-clip selection, though, that will just be start and end times of the selected clip.
             t0 = selectionController()->selectedClipStartTime();
             t1 = selectionController()->selectedClipEndTime();
         } else {
             t0 = selectionController()->dataSelectedStartTime();
-            t1 = selectionController()->dataSelectedEndTime();
```

Comment:
could you please explain why you removed this?

Reply:
Mistake - good catch.